eclass/kernel-install.eclass

```diff
@@ -206,12 +195,68 @@ kernel-install_get_qemu_arch() {
 		arm64)
 			echo aarch64
 			;;
+		ppc64)
+			echo ppc64
+			;;
 		*)
 			die "${FUNCNAME}: unsupported ARCH=${ARCH}"
 			;;
 	esac
 }
 
+# @FUNCTION: kernel-install_create_init
+# @USAGE: <filename>
+# @DESCRIPTION:
+# Create minimal /sbin/init
+kernel-install_create_init() {
+	debug-print-function ${FUNCNAME} "${@}"
+	
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	[[ -z ${1} ]] && die "${FUNCNAME}: empty argument specified"
+
+	local output="${1}"
+	[[ -f ${output} ]] && die "${FUNCNAME}: ${output} already exists"
+
+	cat <<-_EOF_ >"${T}/init.c" || die
+		#include <stdio.h>
+		int main() {
+			printf("Hello, World!\n");
+			return 0;
+		}
+	_EOF_
+
+	$(tc-getBUILD_CC) -Os -static "${T}/init.c" -o "${output}" || die 
+	$(tc-getBUILD_STRIP) "${output}" || die
+}
+
+# @FUNCTION: kernel-install_create_qemu_image
+# @USAGE: <filename>
+# @DESCRIPTION:
+# Create minimal qemu raw image
+kernel-install_create_qemu_image() {
+	debug-print-function ${FUNCNAME} "${@}"
+
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	[[ -z ${1} ]] && die "${FUNCNAME}: empty argument specified"
+
+	local image="${1}"
+	[[ -f ${image} ]] && die "${FUNCNAME}: ${image} already exists"
+
+	local imageroot="${T}/imageroot"
+	[[ -d ${imageroot} ]] && die "${FUNCNAME}: ${imageroot} already exists"
+	mkdir "${imageroot}" || die
+
+	# some layout needed to pass dracut's usable_root() validation
+	mkdir -p "${imageroot}"/{bin,dev,etc,lib,proc,root,sbin,sys} || die
+	touch "${imageroot}/lib/ld-fake.so" || die
+
+	kernel-install_create_init "${imageroot}/sbin/init"
+
+	# image may be smaller if needed
+	truncate -s4M "${image}" || die
+	mkfs.ext4 -v -d "${imageroot}" -L groot "${image}" || die
```

Comment:
I suppose you need to depend on something here.

Reply:
those are parts of @system set, but I can explicitly add `sys-fs/e2fsprogs` and `sys-apps/coreutils`